src/borg/borg-init.c

```diff
@@ -420,6 +421,14 @@ void borg_init(void)
     mem_free(memory_test);
     memory_test = NULL;
 
+    /* HACK this directory isn't automatically created on Windows */
```

Comment:
https://github.com/angband/angband/pull/6141 should ensure that the archive directory is present when running the Windows front end from where it was built.    So, you might drop this part now or wait to drop it after that other pull request has been accepted.

Reply:
...which has now happened.